Game class

juliamazey opened the pull request:
Hey guys, I accidentally tried to merge this with Turing's repository...Anyway, this is all we have been working on so far. We have the set up completed and I think we should be able to start with iteration 2. Please take a look and comment!

juliamazey committed:
Adds teams data file

juliamazey committed:
Adds game class and test

juliamazey committed:
Finishes set up

juliamazey committed:
Finishes set up

m-mrcr commented on lib/game_data.rb:
```diff
@@ -9,7 +9,9 @@ class GameData
     @data_file = []
+    binding.pry
     read(path)
+    merge_hashes
```
Hey, we'd talked about making a change from using the merge_hashes module to not using it here, so we could start seeing the rows as hashes rather than the categories as arrays. Are we still gonna do that?

n-flint commented on test/stat_tracker_test.rb:
```diff
@@ -16,21 +16,22 @@ def setup
     stat_tracker = StatTracker.from_csv(@locations)
 
-    expected_hash = [{"team_id"=>1, "franchiseId"=>23, "shortName"=>"New Jersey", "teamName"=>"Devils", "abbreviation"=>"NJD", "link"=>"/api/v1/teams/1"}, {"team_id"=>4, "franchiseId"=>16, "shortName"=>"Philadelphia", "teamName"=>"Flyers", "abbreviation"=>"PHI", "link"=>"/api/v1/teams/4"}, {"team_id"=>26, "franchiseId"=>14, "shortName"=>"Los Angeles", "teamName"=>"Kings", "abbreviation"=>"LAK", " ...
+    expected_hash = {"team_id"=>[1, 4, 26, 14, 6], "franchiseId"=>[23, 16, 14, 31, 6], "shortName"=>["New Jersey", "Philadelphia", "Los Angeles", "Tampa Bay", "Boston"], "teamName"=>["Devils", "Flyers", "Kings", "Lightning", "Bruins"], "abbreviation"=>["NJD", "PHI", "LAK", "TBL", "BOS"], "link"=>["/api/v1/teams/1", "/api/v1/teams/4", "/api/v1/teams/26", "/api/v1/teams/14", "/api/v1/teams/6"]}
```
Yeah, Martin, we did talk about that. Looks like we're still expecting the categories rather than the individual game information, but that we'd had this before? Not sure why that was changed, so let's fix that.

`expected_hash = [{"team_id"=>1, "franchiseId"=>23, "shortName"=>"New Jersey", "teamName"=>"Devils", "abbreviation"=>"NJD", "link"=>"/api/v1/teams/1"}, {"team_id"=>4, "franchiseId"=>16, "shortName"=>"Philadelphia", "teamName"=>"Flyers", "abbreviation"=>"PHI", "link"=>"/api/v1/teams/4"}, {"team_id"=>26, "franchiseId"=>14, "shortName"=>"Los Angeles", "teamName"=>"Kings", "abbreviation"=>"LAK", "link"=>"/api/v1/teams/26"}, {"team_id"=>14, "franchiseId"=>31, "shortName"=>"Tampa Bay", "teamName"=>"Lightning", "abbreviation"=>"TBL", "link"=>"/api/v1/teams/14"}, {"team_id"=>6, "franchiseId"=>6, "shortName"=>"Boston", "teamName"=>"Bruins", "abbreviation"=>"BOS", "link"=>"/api/v1/teams/6"}]`

m-mrcr replied:
Yeah, that's right. I'll implement that.

m-mrcr committed:
Made that change+

Updated the expected_hash for the teams_data and made sure the test read 'teams_data" not "game_data" again--duplication otherwise. Also changed the expected_hash for game_data. We need to go through and change these for all the other tests/change where we use merge_hashes_module, but this is on the right track. I say let's merge now and we'll deal with the conflicts later--easier to edit in atom than this awful thing. haha

juliamazey reviewed (commented):
Yes, I don't really know why I did that. But thanks for catching it!

n-flint commented:
Hey yeah! Everything looks great! I'll merge it now,.
